Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/connected_components.h

```diff
@@ -232,9 +232,18 @@ connected_components(const PolygonMesh& pmesh,
   FiniteDual finite_dual(dual,
     internal::No_border<PolygonMesh, EdgeConstraintMap>(pmesh, ecmap));
 
-  typename GetFaceIndexMap<PolygonMesh, NamedParameters>::const_type
-    fimap = choose_parameter(get_parameter(np, internal_np::face_index),
-                         get_const_property_map(boost::face_index, pmesh));
+  typedef CGAL::Polygon_mesh_processing::GetMapFromNP<boost::face_index_t,
+      CGAL::dynamic_face_property_t<int>,
+      PolygonMesh, NamedParameters, internal_np::face_index_t> MapGetter;
+
+  MapGetter get_map(boost::face_index_t(),
+                    CGAL::dynamic_face_property_t<int>(),
+                    pmesh, np, internal_np::face_index);
+
+  bool need_init = false;
```

Comment:
the init could be done directly in the `get` I think

Reply:
This is not resolved

Reply:
You did not get the point. The whole block should go away and the map should be initialized without the developer having to write anything (init directly in the get).

Reply:
Ah. More overloads and specializations then.